client/client.go

```diff
@@ -235,7 +237,23 @@ func (c *Client) retryRequest(req *http.Request) (*http.Response, error) {
 	var statusCode int
 	var errMessage string
 
-	for i := 0; i < retries; i++ {
+	// Cache the body content for retries.
+	// This is to allow reuse of the original request for the retries attempts
+	// as the act of reading the body (when doing the httpClient.Do()) closes the
+	// Reader.
+	body, err := ioutil.ReadAll(req.Body)
```

Comment:
It looks like this is causing panics when a body isn't specified like when you are doing a read. This is crashing all the tests

```
== RUN   TestAccACLLifeCycle
--- FAIL: TestAccACLLifeCycle (2.93s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x10d5bdc]

goroutine 20 [running]:
testing.tRunner.func1(0xc4201420f0)
	/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x140bd00, 0x16fa0d0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
io/ioutil.readAll.func1(0xc4200612b0)
	/usr/local/go/src/io/ioutil/ioutil.go:30 +0x104
panic(0x140bd00, 0x16fa0d0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
bytes.(*Buffer).ReadFrom(0xc420270540, 0x0, 0x0, 0x73, 0xc4200993e0, 0xc4200612b8)
	/usr/local/go/src/bytes/buffer.go:205 +0x6c
io/ioutil.readAll(0x0, 0x0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/ioutil/ioutil.go:36 +0xb5
io/ioutil.ReadAll(0x0, 0x0, 0x0, 0x0, 0x0, 0xc420061378, 0x19040f35)
	/usr/local/go/src/io/ioutil/ioutil.go:45 +0x3e
github.com/hashicorp/go-oracle-terraform/client.(*Client).retryRequest(0xc4200b03c0, 0xc420274100, 0x4c0, 0xc4201c0000, 0x7)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/client/client.go:244 +0x91
github.com/hashicorp/go-oracle-terraform/client.(*Client).ExecuteRequest(0xc4200b03c0, 0xc420274100, 0x1706e40, 0x67548e35, 0x2)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/client/client.go:195 +0x39
github.com/hashicorp/go-oracle-terraform/compute.(*Client).executeRequest(0xc420099050, 0x1488d29, 0x6, 0xc4200d68c0, 0x3f, 0x0, 0x0, 0x6e, 0xc4201320e0, 0x6e)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/compute_client.go:73 +0x262
github.com/hashicorp/go-oracle-terraform/compute.(*ResourceClient).deleteResource(0xc420258000, 0x148ac1e, 0xc, 0x13e5ee0, 0xc4200930b0)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/compute_resource_client.go:60 +0x6b
github.com/hashicorp/go-oracle-terraform/compute.(*ACLsClient).DeleteACL(0xc420258000, 0xc420061680, 0x6e, 0xc400000076)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/acl.go:134 +0x43
github.com/hashicorp/go-oracle-terraform/compute.deleteACL(0xc4201420f0, 0xc420258000, 0x148ac1e, 0xc)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/acl_test.go:148 +0x56
panic(0x140bd00, 0x16fa0d0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
io/ioutil.readAll.func1(0xc420061a08)
	/usr/local/go/src/io/ioutil/ioutil.go:30 +0x104
panic(0x140bd00, 0x16fa0d0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
bytes.(*Buffer).ReadFrom(0xc4202703f0, 0x0, 0x0, 0x73, 0xc420099201, 0xc4201ba120)
	/usr/local/go/src/bytes/buffer.go:205 +0x6c
io/ioutil.readAll(0x0, 0x0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/ioutil/ioutil.go:36 +0xb5
io/ioutil.ReadAll(0x0, 0x0, 0x0, 0x0, 0x0, 0xc420061ad0, 0xe61fa6c6)
	/usr/local/go/src/io/ioutil/ioutil.go:45 +0x3e
github.com/hashicorp/go-oracle-terraform/client.(*Client).retryRequest(0xc4200b03c0, 0xc420274000, 0x4c0, 0xc4201c0000, 0x7)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/client/client.go:244 +0x91
github.com/hashicorp/go-oracle-terraform/client.(*Client).ExecuteRequest(0xc4200b03c0, 0xc420274000, 0x1706e40, 0x6752465d, 0x2)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/client/client.go:195 +0x39
github.com/hashicorp/go-oracle-terraform/compute.(*Client).executeRequest(0xc420099050, 0x1488596, 0x3, 0xc4200d6740, 0x3f, 0x0, 0x0, 0x144d580, 0xc4201baa01, 0xc4200b4960)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/compute_client.go:73 +0x262
github.com/hashicorp/go-oracle-terraform/compute.(*ResourceClient).getResource(0xc420258000, 0x148ac1e, 0xc, 0x13d0e20, 0xc4200b4960, 0xc420061dd8, 0x20)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/compute_resource_client.go:45 +0x6f
github.com/hashicorp/go-oracle-terraform/compute.(*ACLsClient).GetACL(0xc420258000, 0xc420061e58, 0xc4201420f0, 0xc420258000, 0x148ac1e)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/acl.go:85 +0x72
github.com/hashicorp/go-oracle-terraform/compute.TestAccACLLifeCycle(0xc4201420f0)
	/Users/mbfrahry/workspace/gocode/src/github.com/hashicorp/go-oracle-terraform/compute/acl_test.go:48 +0x364
testing.tRunner(0xc4201420f0, 0x14a69e8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0
FAIL	github.com/hashicorp/go-oracle-terraform/compute	2.998s
```

Reply:
Resolved.
Also fixed the tests to not attempt to run a couple of the DB Acceptance tests during `make test`